modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/index/meta/io/IndexMetaIo.java

```diff
@@ -44,8 +47,13 @@ public interface IndexMetaIo {
     /** Index tree meta page id offset - long (8 bytes). */
     int INDEX_TREE_META_PAGE_ID_OFFSET = INDEX_ID_LSB_OFFSET + Long.BYTES;
 
+    /** Offset of the inline size in bytes - short (2 bytes). */
+    int INDEX_INLINE_SIZE_OFFSET = INDEX_TREE_META_PAGE_ID_OFFSET + Long.BYTES;
```

Comment:
This is the part that I don't like. We have index meta page. Why do we have to put this information into a meta tree instead of meta page? There will be more flags and stuff in the future, we won't be placing them in the tree.
Please move inline size into the meta page.

Reply:
Try fix it